test/util.spec.js

```diff
@@ -24,125 +24,74 @@ describe('util', () => {
+    // reset decoder to default
+    dagCBOR.util.configureDecoder()
   })
 
-  it('error catching', (done) => {
+  it('error catching', () => {
     const circlarObj = {}
     circlarObj.a = circlarObj
-    dagCBOR.util.serialize(circlarObj, (err, serialized) => {
-      expect(err).to.exist()
-      expect(serialized).to.not.exist()
-      done()
-    })
+    expect(() => dagCBOR.util.serialize(circlarObj)).to.throw(
+      'The object passed has circular reference')
   })
 
-  it('.cid', (done) => {
-    dagCBOR.util.cid(obj, (err, cid) => {
-      expect(err).to.not.exist()
-      expect(cid.version).to.equal(1)
-      expect(cid.codec).to.equal('dag-cbor')
-      expect(cid.multihash).to.exist()
-      const mh = multihash.decode(cid.multihash)
-      expect(mh.name).to.equal('sha2-256')
-      done()
-    })
-  })
-
-  it('.cid with hashAlg', (done) => {
-    dagCBOR.util.cid(obj, { hashAlg: 'sha2-512' }, (err, cid) => {
-      expect(err).to.not.exist()
-      expect(cid.version).to.equal(1)
-      expect(cid.codec).to.equal('dag-cbor')
-      expect(cid.multihash).to.exist()
-      const mh = multihash.decode(cid.multihash)
-      expect(mh.name).to.equal('sha2-512')
-      expect(mh.length).to.equal(64)
-      done()
-    })
+  it('.cid', async () => {
+    const cid = await dagCBOR.util.cid(serializedObj)
+    expect(cid.version).to.equal(1)
+    expect(cid.codec).to.equal('dag-cbor')
+    expect(cid.multihash).to.exist()
+    const mh = multihash.decode(cid.multihash)
+    expect(mh.name).to.equal('sha2-256')
   })
 
-  it('.cid with hashAlg and hashLen', (done) => {
-    dagCBOR.util.cid(obj, { hashAlg: 'keccak-256', hashLen: 28 }, (err, cid) => {
-      expect(err).to.not.exist()
-      expect(cid.version).to.equal(1)
-      expect(cid.codec).to.equal('dag-cbor')
-      expect(cid.multihash).to.exist()
-      const mh = multihash.decode(cid.multihash)
-      expect(mh.name).to.equal('keccak-256')
-      expect(mh.length).to.equal(28)
-      // The CID must be 32 bytes including 4 bytes for
-      // <cid-version><multicodec><hash-function><digest-size>
-      expect(cid.buffer.length).to.equal(32)
-      expect(cid.toBaseEncodedString()).to.equal('z6dSUELEcAsg5oXs7gsv42rYfczTLizSBTpGUa5M3bxe')
-      done()
-    })
-  })
-
-  it('strings', (done) => {
-    dagCBOR.util.cid('some test string', (err, cid) => {
-      expect(err).to.not.exist()
-      expect(cid.version).to.equal(1)
-      expect(cid.codec).to.equal('dag-cbor')
-      expect(cid.multihash).to.exist()
-      done()
-    })
+  it('.cid with hashAlg', async () => {
+    const cid = await dagCBOR.util.cid(serializedObj, { hashAlg: 'sha2-512' })
+    expect(cid.version).to.equal(1)
+    expect(cid.codec).to.equal('dag-cbor')
+    expect(cid.multihash).to.exist()
+    const mh = multihash.decode(cid.multihash)
+    expect(mh.name).to.equal('sha2-512')
+    expect(mh.length).to.equal(64)
   })
 
   it.skip('serialize and deserialize - garbage', (done) => {
```

Comment:
This is an interesting fuzzing test, I think we should fix it up with this work, leave it skipped (because it breaks in the browser) and use it as a way to improve CBOR decoding (with borc or an alternative). #40 disabled it and after playing with it I can see why, borc seems to have a few problems with novel data in the browser.

So, here's a working version that can be put in place, still skipped:

```js
  it.skip('fuzz serialize and deserialize with garbage', () => {
    for (let i = 0; i < 1000; i++) {
      let original = { in: garbage(100) }
      let encoded = dagCBOR.util.serialize(input)
      let decoded = dagCBOR.util.deserialize(encoded)
      expect(decoded).to.eql(original)
    })
  })
```

Add a `console.log(JSON.stringify(original))` to see the type of data borc chokes on in the browser, there's a few types of errors it can encounter.